src/Simplify_Internal.h

```diff
@@ -103,6 +103,14 @@ class Simplify : public VariadicVisitor<Simplify, Expr, Stmt> {
         }
     };
 
+    HALIDE_ALWAYS_INLINE
```

Comment:
I don't think preserving the modulus/remainder is a good idea. I think the body of the if could be `*b = ExprInfo{}`